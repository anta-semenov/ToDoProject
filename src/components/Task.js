import React from 'react'

export default class Task extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className = 'task'>
    
      </div>
    )
  }
}

Task.propTypes = {
  id: React.PropTypes.number.isRequired,
  title: React.PropTypes.string.isRequired,
  completed: React.PropTypes.bool.isRequired,
  today: React.PropTypes.bool.isRequired,

  description: React.PropTypes.string,
  priority: React.PropTypes.string,
  date: React.PropTypes.object
}
