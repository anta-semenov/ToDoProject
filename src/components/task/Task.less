@import '../../styles/variables.less';

.task {
  position: relative;
  margin: 0;
  padding: 0 5px;

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;

  border-bottom: 1px solid @color-border-light;
  transition: @transition-basic;
  cursor: pointer;

  &:first-child {
    border-top: 1px solid @color-border-light;
  }
  &:hover {
    background-color: fade(@color-bg-light-gray, 50%);
  }
  &.is-active {
    background-color: @color-bg-light-gray;
  }
}
.task__body {
  position: relative;
  padding: 10px 5px;
  flex-grow: 1;

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
}
.task__title {
  position: relative;
  flex-shrink: 0;

  font-size: @font-size-basic;
}
