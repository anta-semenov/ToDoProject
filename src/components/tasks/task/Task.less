@import '../../../styles/variables.less';

.task {
  position: relative;
  margin: 0;
  padding: 2px 8px;

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;

  border-bottom: 1px solid @color-border-light-gray;
  border-top: 1px solid transparent;
  transition: @transition-basic;

  &::before {
    //content: '';
    position: absolute;
    z-index: 0;
    width: 0;
    height: 1px;
    left: 32px;
    top: 50%;
    pointer-events: none;
    transition: @transition-basic;
    background-color: transparent;
  }

  &:first-child {
    border-top-color: @color-border-light-gray;
  }
  &:hover {
    background-color: lighten(@color-bg-light-gray, 1%);
  }
  &.is-active {
    background-color: @color-bg-light-gray;
    &:hover {
      background-color: @color-bg-light-gray;
    }
  }
  &.is-completed {
    &::before {
      z-index: 3;
      width: ~"calc(100% - 28px)";
      background-color: fade(@color-red, 50%);
    }
  }
}


.task__priority {
  position: relative;
  z-index: 2;
  box-sizing: border-box;
  margin: 0 10px 0 0;
  padding: 4px;
  width: 22px;
  height: 22px;
  cursor: pointer;
  transform-origin: center left;
  transition: 0.2s ease-out;
  flex-shrink: 0;

  &:hover {
    z-index: 10;
    transform: scale(4.5);
    background-color: #ffffff;
    box-shadow: 0 2px 8px 0 fade(#000000, 20%);
    border-radius: 1px;
    transition: 0.3s ease-out 0.15s;
  }

  .task.is-completed &,
  .task.is-latent-today & {
    pointer-events: none;
    opacity: 0.3;
  }
}
.task__priority-level {
  position: relative;
  z-index: 1;
  height: 2px;
  margin: 0 0 2px 0;
  background-color: @color-border-gray;
  transition: @transition-basic;

  &:hover {
    background-color: @color-border-dark-gray;
  }

  &.task__priority-level--none {
    position: absolute;
    z-index: 0;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: transparent;
  }

  &.task__priority-level--low {
    width: 8px;
    .task__priority.task__priority--low & {
      background-color: @color-accent;
    }
  }

  &.task__priority-level--medium {
    width: 10px;
    .task__priority.task__priority--medium & {
      background-color: @color-accent;
    }
  }

  &.task__priority-level--high {
    width: 12px;
    .task__priority.task__priority--high & {
      background-color: @color-accent;
    }
  }

  &.task__priority-level--max {
    width: 14px;
    .task__priority.task__priority--max & {
      background-color: @color-accent;
    }
  }
}
.task__body {
  position: relative;
  padding: 11px 0 11px 0;
  flex-grow: 1;
  min-width: 0;

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  cursor: pointer;
  transition: opacity @transition-basic;

  .task.is-active & {
    cursor: default;
  }
  .task.is-completed &,
  .task.is-latent-today & {
    opacity: 0.1;
  }
}
.task__main {
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  overflow: hidden;
}
.task__title {
  position: relative;
  flex-shrink: 0;
  max-width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  font-size: @font-size-basic;
}
.task__description {
  position: relative;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  word-wrap: break-word;

  font-size: @font-size-basic;
  font-weight: 300;
  color: @color-font-medium;

  &::before {
    content: 'â€”';
    position: relative;
    margin: 0 8px;
    color: @color-font-light
  }
}

.task__extra {
  position: relative;
  min-width: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  flex-shrink: 0;
}

.task__date {
  position: relative;
  margin: 0 0 0 10px;
  flex-shrink: 0;
  float: right;

  text-align: right;
  font-size: @font-size-small;
  font-weight: 300;
  color: @color-font-medium;
}

.task__tracking {
  position: absolute;
  z-index: 10;
  opacity: 0;
  width: 24px;
  height: 18px;
  top: 50%;
  right: 0;
  margin: 0;
  padding: 0 0 0 6px;
  transform: translateY(-50%);
  border: none;
  background-color: lighten(@color-bg-light-gray, 1%);
  outline: none;
  cursor: pointer;
  transition: opacity @transition-basic;

  &.is-active {
    position: relative;
    transform: none;
    opacity: 1;
    background-color: transparent;

    &::before {
      content: '';
      position: absolute;
      z-index: 2;
      top: 2px;
      right: -1px;
      width: 6px;
      height: 6px;
      border-radius: 2px;
      background-color: #fdfdfd;
      transition: opacity @transition-basic;
    }
    &::after {
      content: '';
      position: absolute;
      z-index: 3;
      top: 3px;
      right: 0;
      width: 4px;
      height: 4px;
      border-radius: 2px;
      background-color: @color-red;
      animation: rec 1s ease-in infinite alternate;
      transition: opacity @transition-basic;
    }
    &:hover::before,
    &:hover::after {
      opacity: 0;
      animation: none;
    }

    .task.is-active & {
      background-color: @color-bg-light-gray;
      &::before {
        background-color: @color-bg-light-gray;
      }
    }
  }

  .task:hover & {
    opacity: 1;
  }

  svg {
    pointer-events: none;
  }
}

.task__tracking svg {
  opacity: 1;
  transition: opacity @transition-basic;
}
.task__tracking.is-active svg {
  opacity: 0.5;
  .task:hover & {
    opacity: 1;
  }
}

.task__tracking svg g[id^='stopwatch__stopwatch-'] {
  opacity: 1;
  transition: opacity @transition-basic;
}
.task__tracking:hover svg g[id^='stopwatch__stopwatch-'] {
  opacity: 0;
}

.task__tracking svg circle[id^='stopwatch__bezel-'] {
  opacity: 1;
  transition: opacity @transition-basic, transform @transition-basic;
  transform-origin: center;
}
.task__tracking:hover svg circle[id^='stopwatch__bezel-'] {
  opacity: 0.5;
  transform: scale(1.25);
}

.task__tracking svg g[id^='stopwatch__stop-'] {
  opacity: 0;
  transition: opacity @transition-basic;
}
.task__tracking.is-active:hover svg g[id^='stopwatch__stop-'] {
  opacity: 1;
}

.task__tracking svg g[id^='stopwatch__start-'],
.task__tracking.is-active svg g[id^='stopwatch__start-'],
.task__tracking.is-active:hover svg g[id^='stopwatch__start-'] {
  opacity: 0;
  transition: opacity @transition-basic;
}
.task__tracking:hover svg g[id^='stopwatch__start-'] {
  opacity: 1;
}
