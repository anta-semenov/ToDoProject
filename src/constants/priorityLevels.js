export const PRIORITY_NONE = 'none'
export const PRIORITY_LOW = 'low'
export const PRIORITY_MEDIUM = 'medium'
export const PRIORITY_HIGH = 'high'
export const PRIORITY_MAX = 'max'
export const PRIORITY = [PRIORITY_NONE, PRIORITY_LOW, PRIORITY_MEDIUM, PRIORITY_HIGH, PRIORITY_MAX]
