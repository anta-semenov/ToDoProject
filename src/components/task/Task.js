import React from 'react'
import PureRenderMixin from 'react-addons-pure-render-mixin'
import { Map } from 'immutable'
import * as priorityLevels from '../../constants/priorityLevels'
import { DATE_FORMAT } from '../../constants/defaults'
import { descriptionToString } from '../../utils/descriptionTransform'

import './Task.less'

export default class Task extends React.Component {
  constructor(props) {
    super(props)
    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this)
  }

  render() {
    return (
      <li className={`task ${this.props.completed ? 'is-completed' : ''} ${this.props.active ? 'is-active' : ''} ${this.props.latentToday ? 'is-latent-today' : ''} `}>
        <input type='checkbox' className='task__completed' checked={this.props.completed} onChange={() => this.props.onTaskCheckboxClick(this.props.id, !this.props.completed)} />
        <div className={`task__today ${this.props.today ? 'is-checked' : ''}`} onClick={() => this.props.onTaskTodayClick(this.props.id, !this.props.today)} />
        <div className={`task__priority task__priority--${this.props.priority ? this.props.priority : 'none'}`} >
          <div className='task__priority-level task__priority-level--none' onClick={() => this.props.onPriorityClick(this.props.id, priorityLevels.PRIORITY_NONE)} />
          <div className='task__priority-level task__priority-level--max' onClick={() => this.props.onPriorityClick(this.props.id, priorityLevels.PRIORITY_MAX)} />
          <div className='task__priority-level task__priority-level--high' onClick={() => this.props.onPriorityClick(this.props.id, priorityLevels.PRIORITY_HIGH)} />
          <div className='task__priority-level task__priority-level--medium' onClick={() => this.props.onPriorityClick(this.props.id, priorityLevels.PRIORITY_MEDIUM)} />
          <div className='task__priority-level task__priority-level--low' onClick={() => this.props.onPriorityClick(this.props.id, priorityLevels.PRIORITY_LOW)} />
        </div>
        <div className='task__body' onClick={() => this.props.onTaskClick(this.props.id)}>
          <div className='task__main'>
            <div className='task__title'>{this.props.title}</div>
            {descriptionToString(this.props.description) ? <div className='task__description'>{descriptionToString(this.props.description)}</div> : null}
          </div>
          {this.props.date ? <div className='task__date'>{this.props.date.toLocaleDateString('en-US', DATE_FORMAT)}</div> : null}
        </div>
      </li>
    )
  }
}

Task.propTypes = {
  id: React.PropTypes.number.isRequired,
  title: React.PropTypes.string.isRequired,
  completed: React.PropTypes.bool.isRequired,
  today: React.PropTypes.bool.isRequired,

  onTaskClick: React.PropTypes.func.isRequired,
  onTaskCheckboxClick: React.PropTypes.func.isRequired,
  onTaskTodayClick: React.PropTypes.func.isRequired,
  onPriorityClick: React.PropTypes.func.isRequired,

  description: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.instanceOf(Map)
  ]),
  priority: React.PropTypes.string,
  date: React.PropTypes.instanceOf(Date),
  active: React.PropTypes.bool,
  latentToday: React.PropTypes.bool
}
