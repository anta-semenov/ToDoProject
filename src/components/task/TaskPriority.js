import React from 'react'
import PureRenderMixin from 'react-addons-pure-render-mixin'
import * as priorityLevels from '../../constants/priorityLevels'

export default class TaskPrority extends React.Component {
  constructor(props) {
    super(props)
    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this)
  }
  render() {
    return (
      <div className={`task__priority task__priority--${this.props.priority ? this.props.priority : 'none'}`} >
        <div className='task__priority-level task__priority-level--none' onClick={() => this.props.onPriorityClick(this.props.taskId, priorityLevels.PRIORITY_NONE)} />
        <div className='task__priority-level task__priority-level--max' onClick={() => this.props.onPriorityClick(this.props.taskId, priorityLevels.PRIORITY_MAX)} />
        <div className='task__priority-level task__priority-level--high' onClick={() => this.props.onPriorityClick(this.props.taskId, priorityLevels.PRIORITY_HIGH)} />
        <div className='task__priority-level task__priority-level--medium' onClick={() => this.props.onPriorityClick(this.props.taskId, priorityLevels.PRIORITY_MEDIUM)} />
        <div className='task__priority-level task__priority-level--low' onClick={() => this.props.onPriorityClick(this.props.taskId, priorityLevels.PRIORITY_LOW)} />
      </div>
    )
  }
}

TaskPrority.propTypes = {
  taskId: React.PropTypes.number.isRequired,
  priority: React.PropTypes.string,
  onPriorityClick: React.PropTypes.func.isRequired
}
