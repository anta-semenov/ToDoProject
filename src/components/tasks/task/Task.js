import React from 'react'
import PureRenderMixin from 'react-addons-pure-render-mixin'
import { Map } from 'immutable'
import { DATE_FORMAT } from '../../../constants/defaults'
import { descriptionToString } from '../../../utils/descriptionTransform'
import Today from '../../elements/today/Today'
import Checkbox from '../../elements/checkbox/Checkbox'
import Priority from '../../elements/priority/Priority'

import './Task.less'

const appearance = 'tasks-list'

export default class Task extends React.Component {
  constructor(props) {
    super(props)
    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this)
  }

  render() {
    return (
      <li className={`task ${this.props.completed ? 'is-completed' : ''} ${this.props.active ? 'is-active' : ''} ${this.props.latent && !this.props.completed ? 'is-latent-today' : ''} `}>
        <Checkbox appearance={appearance} checked={this.props.completed} dimmed={this.props.latent && !this.props.completed} onClick={() => this.props.onTaskCheckboxClick(this.props.id, !this.props.completed)} />
        <Today appearance={appearance} checked={this.props.today} dimmed={this.props.latent && !this.props.completed} disabled={this.props.completed} onClick={() => this.props.onTaskTodayClick(this.props.id, !this.props.today)} />
        <Priority appearance={appearance} priority={this.props.priority} dimmed={this.props.latent && !this.props.completed} disabled={this.props.completed} onClick={(priority) => this.props.onPriorityClick(this.props.id, priority)}/>
        <div className='task__body' onClick={() => this.props.onTaskClick(this.props.id)}>
          <div className='task__main'>
            <div className='task__title'>{this.props.title}</div>
            {descriptionToString(this.props.description) ? <div className='task__description'>{descriptionToString(this.props.description)}</div> : null}
          </div>
          {this.props.date ? <div className='task__date'>{this.props.date.toLocaleDateString('en-US', DATE_FORMAT)}</div> : null}
        </div>
      </li>
    )
  }
}

Task.propTypes = {
  id: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  completed: React.PropTypes.bool.isRequired,
  today: React.PropTypes.bool.isRequired,

  onTaskClick: React.PropTypes.func.isRequired,
  onTaskCheckboxClick: React.PropTypes.func.isRequired,
  onTaskTodayClick: React.PropTypes.func.isRequired,
  onPriorityClick: React.PropTypes.func.isRequired,

  description: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.instanceOf(Map)
  ]),
  priority: React.PropTypes.string,
  date: React.PropTypes.instanceOf(Date),
  active: React.PropTypes.bool,
  latent: React.PropTypes.bool
}
