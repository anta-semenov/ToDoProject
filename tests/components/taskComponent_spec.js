import React from 'react'
import { fromJS } from 'immutable'
import { shallow, mount } from 'enzyme'
import { TaskClass as Task } from '../../src/components/tasks/task/Task'
import * as priorityLevels from '../../src/constants/priorityLevels'

const testTasks = fromJS([
  {
    id: 'b41sogy3s0o0',
    title: 'Explore React',
    completed: true,
    today: true,
    description: 'Make Tutorial on oficial react website',
    priority: priorityLevels.PRIORITY_MEDIUM,
    date: new Date(2015, 4, 15),
    connectDragPreview: element => element,
    connectDragSource: element => element
  },
  {
    id: 'b41sogy3s0o1',
    title: 'Explore Redux',
    completed: false,
    today: false,
    connectDragPreview: element => element,
    connectDragSource: element => element
  }
])

const mockDragFunctions = {
  connectDragPreview: element => element,
  connectDragSource: element => element
}
const props = {
  id: '37897hjkdha',
  title: 'test title',
  someday: true,
  onTaskClick: () => {},
  onTaskCheckboxClick: () => {},
  onTaskTodayClick: () => {},
  onTaskPriorityClick: () => {},
  onTaskTrackingClick: () => {},
  addTaskToProject: () => {},
  addTaskContext: () => {}
}

describe('Task component', () => {
  it('Should render Task component with active and completed classes', () => {
    const taskComponent = shallow(
      <Task {...props} active={true} completed={true} {...mockDragFunctions} />
    )
    expect(taskComponent).toMatchSnapshot()
  })
  it('Should render Task component without active and completed classes', () => {
    const taskComponent = shallow(
      <Task {...props} active={false} completed={false} {...mockDragFunctions} />
    )
    expect(taskComponent).toMatchSnapshot()
  })
  describe('Checkbox', () => {
    const checkboxClass = '.checkbox'
    it('Should render complete checkbox', () => {
      const taskComponent = mount(
        <Task {...props} completed={testTasks.get(0).get('completed')} {...mockDragFunctions} />
      )
      const checkboxComponent = taskComponent.find(checkboxClass)
      expect(checkboxComponent).toMatchSnapshot()
    })
    it('Should render uncomplete checkbox', () => {
      const taskComponent = mount(
        <Task {...props} completed={testTasks.get(1).get('completed')} {...mockDragFunctions} />
      )
      const checkboxComponent = taskComponent.find(checkboxClass)
      expect(checkboxComponent).toMatchSnapshot()
    })
    it('Should invoke complete callback when change event occurs', () => {
      // const checkboxClass = 'checkbox'
      let checkedId = -12
      let checkStatus = true
      const callback = (id, status) => {
        checkedId = id
        checkStatus = status
      }
      const taskComponent = mount(
        <Task
          {...props}
          completed={testTasks.getIn([0, 'completed'])}
          id={testTasks.getIn([0, 'id'])}
          onTaskCheckboxClick={callback}
          {...mockDragFunctions}
        />
      )
      const checkboxComponent = taskComponent.find(checkboxClass).first()
      expect(checkedId).toBe(-12)
      expect(checkStatus).toBe(true)

      checkboxComponent.simulate('click')
      expect(checkedId).toBe('b41sogy3s0o0')
      expect(checkStatus).toBe(false)
    })
  })
  describe('Today', () => {
    const todayClass = '.today'
    it('Should render checked today toggle', () => {
      const taskComponent = mount(
        <Task {...props} today={testTasks.get(0).get('today')} {...mockDragFunctions} />
      )
      const checkboxComponent = taskComponent.find(todayClass)
      expect(checkboxComponent).toMatchSnapshot()
    })
    it('Should render unchecked today toggle', () => {
      const taskComponent = mount(
        <Task {...props} today={testTasks.get(1).get('today')} {...mockDragFunctions} />
      )
      const checkboxComponent = taskComponent.find(todayClass)
      expect(checkboxComponent).toMatchSnapshot()
    })
    it('Should invoke today callback when click occurs', () => {
      let callbackId = -12
      let callbackStatus = true
      const callback = (id, status) => {
        callbackId = id
        callbackStatus = status
      }
      const taskComponent = mount(
        <Task
          {...props}
          today={testTasks.get(0).get('today')}
          id={testTasks.get(0).get('id')}
          onTaskTodayClick={callback}
          {...mockDragFunctions}
        />
      )

      expect(callbackId).toBe(-12)
      expect(callbackStatus).toBe(true)

      taskComponent.find(todayClass).simulate('click')
      expect(callbackId).toBe(testTasks.get(0).get('id'))
      expect(callbackStatus).toBe(!testTasks.get(0).get('today'))
    })
  })
  describe('Priority', () => {
    it('Should invoke callback when click on priority level', () => {
      let priority = ''
      let callbackId = -12
      const callback = (id, priorityLevel) => {
        callbackId = id
        priority = priorityLevel
      }
      const priorityLevelNoneClass = '.priority-level--none'
      const priorityLevelLowClass = '.priority-level--low'
      const priorityLevelMediumClass = '.priority-level--medium'
      const priorityLevelHighClass = '.priority-level--high'
      const priorityLevelMaxClass = '.priority-level--max'

      const taskComponentNone = mount(
        <Task
          {...props}
          id={'b41sogy3s0o0'}
          onTaskPriorityClick={callback}
          {...mockDragFunctions}
        />
      )
      const taskComponentLow = mount(
        <Task
          {...props}
          id={'b41sogy3s0o1'}
          onTaskPriorityClick={callback}
          {...mockDragFunctions}
        />
      )
      const taskComponentMedium = mount(
        <Task
          {...props}
          id={'b41sogy3s0o2'}
          onTaskPriorityClick={callback}
          {...mockDragFunctions}
        />
      )
      const taskComponentHigh = mount(
        <Task
          {...props}
          id={'b41sogy3s0o3'}
          onTaskPriorityClick={callback}
          {...mockDragFunctions}
        />
      )
      const taskComponentMax = mount(
        <Task
          {...props}
          id={'b41sogy3s0o4'}
          onTaskPriorityClick={callback}
          {...mockDragFunctions}
        />
      )

      const priorityLevelComponentNone = taskComponentNone.find(priorityLevelNoneClass)
      const priorityLevelComponentLow = taskComponentLow.find(priorityLevelLowClass)
      const priorityLevelComponentMedium = taskComponentMedium.find(priorityLevelMediumClass)
      const priorityLevelComponentHigh = taskComponentHigh.find(priorityLevelHighClass)
      const priorityLevelComponentMax = taskComponentMax.find(priorityLevelMaxClass)

      expect(priority).toBe('')
      expect(callbackId).toBe(-12)

      priorityLevelComponentNone.simulate('click')
      expect(priority).toBe(priorityLevels.PRIORITY_NONE)
      expect(callbackId).toBe('b41sogy3s0o0')

      priorityLevelComponentLow.simulate('click')
      expect(priority).toBe(priorityLevels.PRIORITY_LOW)
      expect(callbackId).toBe('b41sogy3s0o1')

      priorityLevelComponentMedium.simulate('click')
      expect(priority).toBe(priorityLevels.PRIORITY_MEDIUM)
      expect(callbackId).toBe('b41sogy3s0o2')

      priorityLevelComponentHigh.simulate('click')
      expect(priority).toBe(priorityLevels.PRIORITY_HIGH)
      expect(callbackId).toBe('b41sogy3s0o3')

      priorityLevelComponentMax.simulate('click')
      expect(priority).toBe(priorityLevels.PRIORITY_MAX)
      expect(callbackId).toBe('b41sogy3s0o4')
    })
  })
  describe('Task Body', () => {
    const titleClass = '.task__title'
    it('Should render task title', () => {
      const taskComponent = mount(
        <Task {...props} title={testTasks.get(0).get('title')} {...mockDragFunctions} />
      )
      const titleComponent = taskComponent.find(titleClass)
      expect(titleComponent).toMatchSnapshot()
    })
    it('Should render task description', () => {
      const taskComponent = shallow(
        <Task {...props} description={testTasks.get(0).get('description')} {...mockDragFunctions} />
      )
      expect(taskComponent.find('.task__description')).toMatchSnapshot()
    })
    it('Should render task date', () => {
      const taskComponent = shallow(
        <Task {...props} date={testTasks.get(0).get('date')} {...mockDragFunctions} />
      )
      expect(taskComponent.find('.task__date')).toMatchSnapshot()
    })
    it('Should invoke callback when click on task body, title, description or date occurs', () => {
      let callbackId = -12
      const callback = id => (callbackId = id)

      const taskComponent = mount(
        <Task
          {...props}
          id={'b41sogy3s0o0'}
          title={testTasks.get(0).get('title')}
          description={testTasks.get(0).get('description')}
          date={testTasks.get(0).get('date')}
          onTaskClick={callback}
          {...mockDragFunctions}
        />
      )
      const bodyComponent = taskComponent.find('.task__body')
      const titleComponent = taskComponent.find('.task__title')
      const descriptionComponent = taskComponent.find('.task__description')

      expect(callbackId).toBe(-12)
      bodyComponent.simulate('click')
      expect(callbackId).toBe('b41sogy3s0o0')

      callbackId = -12
      expect(callbackId).toBe(-12)
      titleComponent.simulate('click')
      expect(callbackId).toBe('b41sogy3s0o0')

      callbackId = -12
      expect(callbackId).toBe(-12)
      descriptionComponent.simulate('click')
      expect(callbackId).toBe('b41sogy3s0o0')
    })
  })
})
