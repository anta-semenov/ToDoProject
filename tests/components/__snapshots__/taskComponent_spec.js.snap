// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component Checkbox Should render complete checkbox 1`] = `
<div
  className="checkbox checkbox--tasks-list is-checked"
  onClick={[Function]}
>
  <svg
    className="checkbox__checkmark"
    viewBox="0 0 17 14"
  >
    <path
      className="checkbox__checkmark-path"
      d="M15.0328152,0.000721764681 C13.8445596,0.691568051 12.2970639,2.12852833 11.2746114,3.04044543 C9.22970639,4.86427962 7.26770294,6.77101537 5.33333333,8.70538498 L1.46459413,5.3340551 L-1.5720758e-13,6.77101537 C2.12780656,8.89882194 4.42141623,11.1924316 6.30051813,13.5689428 C8.70466321,9.61730208 12.6286701,3.95236252 16,0.967906566 C15.8894646,0.746835754 15.4196891,-0.0269120868 15.0328152,0.000721764681 Z"
    />
  </svg>
</div>
`;

exports[`Task component Checkbox Should render uncomplete checkbox 1`] = `
<div
  className="checkbox checkbox--tasks-list"
  onClick={[Function]}
>
  <svg
    className="checkbox__checkmark"
    viewBox="0 0 17 14"
  >
    <path
      className="checkbox__checkmark-path"
      d="M15.0328152,0.000721764681 C13.8445596,0.691568051 12.2970639,2.12852833 11.2746114,3.04044543 C9.22970639,4.86427962 7.26770294,6.77101537 5.33333333,8.70538498 L1.46459413,5.3340551 L-1.5720758e-13,6.77101537 C2.12780656,8.89882194 4.42141623,11.1924316 6.30051813,13.5689428 C8.70466321,9.61730208 12.6286701,3.95236252 16,0.967906566 C15.8894646,0.746835754 15.4196891,-0.0269120868 15.0328152,0.000721764681 Z"
    />
  </svg>
</div>
`;

exports[`Task component Should render Task component with active and completed classes 1`] = `
<li
  className="task is-completed is-active"
>
  <Checkbox
    appearance="tasks-list"
    checked={true}
    dimmed={false}
    onChange={[Function]}
  />
  <Today
    appearance="tasks-list"
    checked={false}
    disabled={true}
    onClick={[Function]}
  />
  <Priority
    appearance="tasks-list"
    disabled={true}
    onClick={[Function]}
  />
  <div
    className="task__body"
    onClick={[Function]}
  >
    <div
      className="task__main"
    >
      <div
        className="task__title"
      >
        test title
      </div>
    </div>
    <div
      className="task__extra"
    />
  </div>
  <Stopwatch
    appearance="tasks-list"
    disabled={true}
    onClick={[Function]}
  />
</li>
`;

exports[`Task component Should render Task component without active and completed classes 1`] = `
<li
  className="task"
>
  <Checkbox
    appearance="tasks-list"
    checked={false}
    dimmed={false}
    onChange={[Function]}
  />
  <Today
    appearance="tasks-list"
    checked={false}
    disabled={false}
    onClick={[Function]}
  />
  <Priority
    appearance="tasks-list"
    disabled={false}
    onClick={[Function]}
  />
  <div
    className="task__body"
    onClick={[Function]}
  >
    <div
      className="task__main"
    >
      <div
        className="task__title"
      >
        test title
      </div>
    </div>
    <div
      className="task__extra"
    />
  </div>
  <Stopwatch
    appearance="tasks-list"
    disabled={false}
    onClick={[Function]}
  />
</li>
`;

exports[`Task component Task Body Should render task date 1`] = `
<div
  className="task__date"
>
  May 15, 2015
</div>
`;

exports[`Task component Task Body Should render task description 1`] = `
<div
  className="task__description"
>
  Make Tutorial on oficial react website
</div>
`;

exports[`Task component Task Body Should render task title 1`] = `
<div
  className="task__title"
>
  Explore React
</div>
`;

exports[`Task component Today Should render checked today toggle 1`] = `
<div
  className="today today--tasks-list is-checked"
  onClick={[Function]}
>
  <svg
    className="today__svg"
    viewBox="0 0 26 26"
  >
    <circle
      className="today__beams"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
    <circle
      className="today__beams--active"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
  </svg>
</div>
`;

exports[`Task component Today Should render unchecked today toggle 1`] = `
<div
  className="today today--tasks-list"
  onClick={[Function]}
>
  <svg
    className="today__svg"
    viewBox="0 0 26 26"
  >
    <circle
      className="today__beams"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
    <circle
      className="today__beams--active"
      cx="13"
      cy="13"
      fill="none"
      r="13"
    />
  </svg>
</div>
`;
